Feature/navigation details screens

haideraljawasim opened the pull request:
# What’s Done
- Implemented the Details screens ui 
- Implemented navigation for the Details screen

AhmadKharfan committed:
add simple demo actor screen

AhmadKharfan committed:
add actor view model

AhmadKharfan committed:
implement actor screen and actor viewmodel

AhmadKharfan committed:
add actor ui screen model

haideraljawasim committed:
make base view model public

haideraljawasim committed:
add SeriesViewModel

haideraljawasim committed:
add sereis interaction listiner

haideraljawasim committed:
add series screen effercts

haideraljawasim committed:
add SeriesScreen interaction listiner

haideraljawasim committed:
add Episode Card component for series screen

shahad-97 committed:
add reviewcardscreen

haideraljawasim committed:
fix a bug in ImageSlider

haideraljawasim committed:
getting images and trailr without launage

haideraljawasim committed:
add loading component to desgin system

hagarabobakr committed:
add dependencies for media details presentation

This commit adds a number of dependencies to the `media_details/presentation` module. These include dependencies for modules like `:domain:vod` and `:image_viewer`, AndroidX libraries, Compose libraries, Coil for image loading, Koin for dependency injection, KotlinX datetime, and various testing libraries such as JUnit, MockK, KotlinX Coroutines Test, Truth, and Turbine. The JUnit Platform is also configured for test tasks.

haideraljawasim committed:
add season tap for series screen

haideraljawasim committed:
get tv series trailr

haideraljawasim committed:
add localzation for sereis screen

haideraljawasim committed:
add the limit of images inside combinin object for images count

shahad-97 committed:
fix code

shahad-97 committed:
add  empty reviews state

haideraljawasim committed:
add the base routs for deatils navagtion screen

hagarabobakr committed:
add new string resource for unsuitable image

The string "unsuitable_image" has been added to the string resources in both English and Arabic.

hagarabobakr committed:
add movie placeholder image to design system module

Adds a new movie placeholder image to the design system module, with dark and light variants for different screen densities (mdpi, hdpi, xhdpi, xxhdpi, xxxhdpi).

hagarabobakr committed:
add movie categories screen and UI models

shahad-97 committed:
add add strings

shahad-97 committed:
delete modifier

haideraljawasim committed:
add Details NavHost

haideraljawasim committed:
split the code into sparte component

haideraljawasim committed:
sprate SereisTopScreen into sprate component

haideraljawasim committed:
refactor screen code to use splited compnent

hagarabobakr committed:
add images

hagarabobakr committed:
Merge branch 'refs/heads/development' into feature/implement-movie-categories-screen

Radwa85 committed:
add presentation layer mappers

Radwa85 committed:
add presentation layer ui models

Radwa85 committed:
add version to `koin-androidx-compose` in `libs.versions.toml`

Radwa85 committed:
add `MovieDetailsUiState` to presentation layer

Radwa85 committed:
add `MovieDetailsUiState` to presentation layer

Radwa85 committed:
add movie details screen

Radwa85 committed:
add movie details view model

Radwa85 committed:
add `MovieDetailsScreenInteractionListener`

Radwa85 committed:
add kotlinx-datetime and koin dependencies to media details feature

Radwa85 committed:
add `MovieDetailsUiEffect` to movie details feature

Radwa85 committed:
Merge remote-tracking branch 'origin/development' into feature/movie-details-screen

shahad-97 committed:
refactor and fix

haideraljawasim committed:
add epsiode details screen effects

haideraljawasim committed:
add interaction listener for epsiode details

haideraljawasim committed:
add EsipdeDetailsInteraction ui state

haideraljawasim committed:
implement Epsisode Details View Model

haideraljawasim committed:
impelment EsiodeDetailScreen

haideraljawasim committed:
extect code into sprate component

haideraljawasim committed:
add res for epsiode details

haideraljawasim committed:
rename SereisTopScreen to SereiesHeaderSection

haideraljawasim committed:
rename TrailerButton to BottomContainer

haideraljawasim committed:
add LoadingIndcator to desgin system

haideraljawasim committed:
refactor sereis screen code

haideraljawasim committed:
make the path of actor image dto  nullable

Radwa85 committed:
add `VideoDto` and `VideoResponseDto`

Radwa85 committed:
rename searchModule to viewModelModule

Radwa85 committed:
replace Text with TextButton in OverviewSection

Radwa85 committed:
add `trailerUrl` property to `Movie` and `MovieDetailsUiModel`

Radwa85 committed:
add movie trailer to repository

Radwa85 committed:
handle watch trailer button click

Radwa85 committed:
update movie details screen

Radwa85 committed:
remove unused variable in `MovieDetailsViewModel`

Radwa85 committed:
use movie details screen for `MainActivity` and rename search module to `viewModelModule`

Radwa85 committed:
add `fetchMovieTrailerUrl` method to `MovieDetailsRemoteDataSource`

Radwa85 committed:
make `BaseViewModel` access modifier public

Radwa85 committed:
remove GetTvSeriesTrailerUseCase from movie

Radwa85 committed:
add GetMovieTrailerUseCase to domainModule

Radwa85 committed:
add media details presentation dependency

AhmadKharfan committed:
add dependencies for media details and series

AhmadKharfan committed:
add necessary dependencies for feature media details

AhmadKharfan committed:
inject viewmodel to domain module

AhmadKharfan committed:
remove temp search module

AhmadKharfan committed:
implement the actor screen with test data

AhmadKharfan committed:
mark movie cast credit as nullable in actor movie cast dto

AhmadKharfan committed:
improve robustness of local date parsing

AhmadKharfan committed:
handle nullable cast in actor repository impl

AhmadKharfan committed:
fix actor biography nullability

AhmadKharfan committed:
add movie details screen related strings

AhmadKharfan committed:
remove all presentation search components and screens

AhmadKharfan committed:
implement media details components

AhmadKharfan committed:
implement actor screen effects

AhmadKharfan committed:
crete ui models for actor screen

AhmadKharfan committed:
add actors screen interactions listener

AhmadKharfan committed:
add acto view model to fetch actor details

AhmadKharfan committed:
introduce base view model to avoid boilerplate in view models

AhmadKharfan committed:
add basic navigation to media details feature

AhmadKharfan committed:
implement actor details screen

AhmadKharfan committed:
implement top actor moveis screen

AhmadKharfan committed:
temp local commit

AhmadKharfan committed:
add new icons for search feature

AhmadKharfan committed:
reinstate search module with filter view model and search view model

AhmadKharfan committed:
change fill alpha to 1 and reorder strings

AhmadKharfan committed:
add media details icons, images and strings

AhmadKharfan committed:
add search screen and filter bottom sheet

AhmadKharfan committed:
remove unused drawables and strings

AhmadKharfan committed:
add placeholders for movie posters

AhmadKharfan committed:
move actor screen components to new package structure

AhmadKharfan committed:
move actor view model to actors screen

AhmadKharfan committed:
implement actor details screen components

AhmadKharfan committed:
fix incorrect package names in top movie screen

AhmadKharfan committed:
refactor actor screen to use novix scaffold and improve overall structure

AhmadKharfan committed:
implement wavy progress indicator for loading state

AhmadKharfan committed:
refactor actor screen and update design system dependencies

haideraljawasim committed:
fix overview top padding

haideraljawasim committed:
Merge branch 'development' into feature/navigation-details-screens

haideraljawasim committed:
Merge branch 'feature/series-screen' into feature/navigation-details-screens

# Conflicts:
#	data/remote_data_source/series/src/main/java/com/sanaa/series/RemoteTvSeriesDataSourceImpl.kt

AhmadKharfan committed:
add icons to actor info card

AhmadKharfan committed:
refactor actor screen to include gallery, top movies, and top series sections

AhmadKharfan committed:
enhance actor screen ui state by adding lifespan

AhmadKharfan committed:
add image placeholder in media details screen

AhmadKharfan committed:
update top movies screen ui and component styling

AhmadKharfan committed:
implement actor gallery screen and top series screen

AhmadKharfan committed:
implement gallery, top series, and gallery card screens

AhmadKharfan committed:
add location and birthday cake icons

haideraljawasim committed:
add review screen ui state

haideraljawasim committed:
implement ReviewViewModel

haideraljawasim committed:
Merge branch 'feature/review_card_screen' into feature/navigation-details-screens

AhmadKharfan committed:
add arabic translations for media details

AhmadKharfan committed:
refactor nav host

haideraljawasim committed:
change navagtion commpose to 2.9.1 in toml file

haideraljawasim committed:
Merge remote-tracking branch 'refs/remotes/origin/feature/151-actor-details-screen-ui' into feature/navigation-details-screens

# Conflicts:
#	app/build.gradle.kts
#	app/src/main/java/com/sanaa/novix/di/appModule.kt
#	app/src/main/java/com/sanaa/novix/di/domainModule.kt
#	feature/media_details/presentation/build.gradle.kts
#	feature/media_details/presentation/src/main/java/com/sanaa/presentation/component/DotSeparator.kt
#	feature/media_details/presentation/src/main/java/com/sanaa/presentation/component/IconWithText.kt
#	feature/media_details/presentation/src/main/java/com/sanaa/presentation/component/ImageSlider.kt
#	feature/media_details/presentation/src/main/java/com/sanaa/presentation/component/InfoSection.kt
#	feature/media_details/presentation/src/main/java/com/sanaa/presentation/component/OverviewSection.kt
#	feature/media_details/presentation/src/main/java/com/sanaa/presentation/navigation/DetailsNavHost.kt
#	feature/media_details/presentation/src/main/java/com/sanaa/presentation/navigation/detailsRouts.kt
#	feature/media_details/presentation/src/main/res/drawable/icon_calender.xml
#	feature/media_details/presentation/src/main/res/drawable/icon_duration.xml
#	feature/media_details/presentation/src/main/res/drawable/icon_seasons.xml
#	feature/media_details/presentation/src/main/res/values/strings.xml

haideraljawasim committed:
,resolve conflict mergeing with Actor screen branch

hagarabobakr committed:
add RequestToLoginBottomSheet to common component

Radwa85 committed:
use novix scaffold

Radwa85 committed:
add`NavigateToReviewsScreen`

Radwa85 committed:
update toCastUiModel mapper

Radwa85 committed:
move OnBlurContent to design system module

Radwa85 committed:
Merge remote-tracking branch 'origin/development' into feature/movie-details-screen

haideraljawasim committed:
add the navigation routs for details screen

haideraljawasim committed:
add ui moudles

haideraljawasim committed:
refactor split the code into components

haideraljawasim committed:
add navigation between screens

haideraljawasim committed:
add strings to string res

haideraljawasim committed:
add review avter image path

haideraljawasim committed:
refactor tv sereis entity the overview to be null

haideraljawasim committed:
fix a bug in Geting reviewes

Radwa85 committed:
use image viewer

Radwa85 committed:
add text color to `IconWithText` component

Radwa85 committed:
add unsuitable image string and placeholders

Radwa85 committed:
add image viewer module to media details feature

hagarabobakr committed:
update Genre enum to include value and fromValue factory method

hagarabobakr committed:
remove unnecessary "please login" images

hagarabobakr committed:
add profile icon for media details

hagarabobakr committed:
update login bottom sheet UI and add new login images

hagarabobakr committed:
rename MoviesByCategoryDto to MoviesByCategoryResponse and update usages

hagarabobakr committed:
rename MoviesByCategoryDto to MoviesByCategoryResponse and update usages

hagarabobakr committed:
enhance MovieCategoriesUiState and MovieCardUiModel

hagarabobakr committed:
add interaction listener to MovieCategoriesScreen

hagarabobakr committed:
add MovieCategoriesEffects

hagarabobakr committed:
move MovieCategoriesScreen to movie_categories package

hagarabobakr committed:
Add ViewModel for movie categories screen

Radwa85 committed:
add new string resources

Radwa85 committed:
organize movie details screen components

Radwa85 committed:
organize movie details screen components

haideraljawasim committed:
Merge remote-tracking branch 'refs/remotes/origin/feature/movie-details-screen' into feature/navigation-details-screens

# Conflicts:
#	app/build.gradle.kts
#	app/src/main/java/com/sanaa/novix/MainActivity.kt
#	app/src/main/java/com/sanaa/novix/NovixApp.kt
#	app/src/main/java/com/sanaa/novix/di/domainModule.kt
#	feature/media_details/presentation/build.gradle.kts
#	feature/media_details/presentation/src/main/java/com/sanaa/presentation/component/IconWithText.kt
#	feature/media_details/presentation/src/main/java/com/sanaa/presentation/component/OverviewSection.kt
#	feature/media_details/presentation/src/main/java/com/sanaa/presentation/details_base/BaseViewModel.kt
#	feature/media_details/presentation/src/main/res/values-ar/strings.xml
#	feature/media_details/presentation/src/main/res/values/strings.xml

Radwa85 committed:
add `GetMovieTrailerUseCase`

haideraljawasim committed:
Merge remote-tracking branch 'origin/feature/movie-details-screen' into feature/navigation-details-screens

haideraljawasim committed:
merging with MovieDetailsScreen

haideraljawasim committed:
Merge remote-tracking branch 'origin/feature/implement-movie-categories-screen' into feature/navigation-details-screens

# Conflicts:
#	data/repositories/movies/src/main/java/com/sanaa/movies/dataSource/remote/MovieDetailsRemoteDataSource.kt
#	feature/media_details/presentation/build.gradle.kts
#	feature/media_details/presentation/src/main/res/values-ar/strings.xml
#	feature/media_details/presentation/src/main/res/values/strings.xml

haideraljawasim committed:
add rate bottom sheet for series detaisl

Radwa85 committed:
fix rating color and genres layout

Radwa85 committed:
remove `belongsToCollection` field from `MovieDetailsDto`

Radwa85 committed:
use `FlowRow` for genres in series header

Radwa85 committed:
fix nullable movieDetails in MovieDetailsViewModel

haideraljawasim committed:
refactor screens code

haideraljawasim committed:
Merge remote-tracking branch 'origin/feature/movie-details-screen' into feature/navigation-details-screens

# Conflicts:
#	feature/media_details/presentation/src/main/java/com/sanaa/presentation/screen/movie_details/MovieDetailsScreen.kt

haideraljawasim committed:
add nivagte between top sereis and movie screens

AhmadKharfan committed:
refactor use case management in domain module

AhmadKharfan committed:
handle nullable cast list in actor remote data source impl test

AhmadKharfan committed:
refactor tests for movie reviews and actor's top movies

AhmadKharfan committed:
include missing profile_path in api response for series credits

AhmadKharfan committed:
fix tests for movie repository and mappers

AhmadKharfan committed:
fix tests for movie and series repositories

AhmadKharfan committed:
add manage actor details use case

AhmadKharfan committed:
add manage episode details use case

AhmadKharfan committed:
add manage episode details use case

AhmadKharfan committed:
add manage episode details use case

AhmadKharfan committed:
add unit tests for the manage actor usecase class

AhmadKharfan committed:
add unit tests for the manage mo usecase class

AhmadKharfan committed:
add unit tests for the manage episode usecase class

AhmadKharfan committed:
add unit tests for the manage episode usecase class

AhmadKharfan committed:
remove tests for removed usecases

AhmadKharfan committed:
refactor view models to use consolidated usecases

ZainabAlnajjar commented on app/src/main/java/com/sanaa/novix/NovixApp.kt:
```diff
@@ -6,7 +6,8 @@ import com.sanaa.novix.di.appModule
 import com.sanaa.novix.di.loggingModule
-import com.sanaa.novix.di.searchModule
+import com.sanaa.novix.di.viewModelModule
+//import com.sanaa.novix.di.searchModule
```
You can delete this comment

ZainabAlnajjar commented on app/src/main/java/com/sanaa/novix/NovixApp.kt:
```diff
@@ -20,7 +21,7 @@ class NovixApp : Application(), KoinComponent {
         startKoin {
             androidContext(this@NovixApp)
-            modules(appModule, searchModule, firebaseModule, domainModule, loggingModule)
+            modules(appModule, viewModelModule, firebaseModule, domainModule, loggingModule)
```
I think we should have seperated the di also by feature, not putting them in the same di module🤔

ZainabAlnajjar commented on feature/media_details/presentation/src/main/java/com/sanaa/presentation/screen/movie_details/components/MoreLikeThisSection.kt:
```diff
@@ -0,0 +1,55 @@
+import com.sanaa.presentation.R as presentationR
+
+@Composable
+fun MoreLikeThisSection(
```
I think `SimilarMoviesSection` could have been better and clearer name , despite what the title in the design

ZainabAlnajjar commented on feature/media_details/presentation/src/main/java/com/sanaa/presentation/screen/movie_details/components/MoreLikeThisSection.kt:
```diff
@@ -0,0 +1,55 @@
+fun MoreLikeThisSection(
+    similarMovies: List<MovieUiModel>,
+    onBookmarkClick: (Int) -> Unit,
+    onSimilarMovieClick: (Int) -> Unit,
```
No need for `Similar` word it is redundant

ZainabAlnajjar commented on feature/media_details/presentation/src/main/java/com/sanaa/presentation/screen/movie_details/components/MoreLikeThisCard.kt:
```diff
@@ -0,0 +1,66 @@
+import com.sanaa.presentation.R as presentationR
+
+@Composable
+fun MoreLikeThisCard(
```
The name is not clear, also this card is general for both movies and tvshow in any page, we can name it `MovieCard` if you want them separated

ZainabAlnajjar commented on feature/media_details/presentation/src/main/java/com/sanaa/presentation/screen/movie_details/MovieDetailsUiState.kt:
```diff
@@ -0,0 +1,14 @@
+import com.sanaa.presentation.model.ActorUiModel
+import com.sanaa.presentation.model.MovieUiModel
+
+data class MovieDetailsUiState(
```
Please rename to `MovieDetailsScreenUiState` for more clarity

ZainabAlnajjar commented on feature/media_details/presentation/src/main/java/com/sanaa/presentation/screen/movie_details/MovieDetailsUiEffect.kt:
```diff
@@ -0,0 +1,9 @@
+package com.sanaa.presentation.screen.movie_details
+
+sealed class MovieDetailsUiEffect {
```
Same here, please add `Screen` to the name

ZainabAlnajjar commented on feature/media_details/presentation/src/main/java/com/sanaa/presentation/screen/movie_categories/MovieCategoriesUiState.kt:
```diff
@@ -0,0 +1,49 @@
+    val rating: Float = 0.0f,
+)
+
+val GenreDisplayNameMap = mapOf(
```
This would just map the genres to English, please refare to the mapping method in the search feature, later we should refactor this to get date from api, then we won't need these mappers, just requesting the categories from api

ZainabAlnajjar commented on feature/media_details/presentation/src/main/java/com/sanaa/presentation/screen/movie_categories/MovieCategoriesUiState.kt:
```diff
@@ -0,0 +1,49 @@
+import androidx.compose.material3.BottomAppBarState
+import entity.Genre
+
+data class MovieCategoriesScreenUiState(
```
The name is confusing, I think you meant `CategoryMovies`

ZainabAlnajjar commented on feature/media_details/presentation/src/main/java/com/sanaa/presentation/screen/movie_categories/MovieCategoriesViewModel.kt:
```diff
@@ -0,0 +1,70 @@
+import details.usecase.movie.GetMoviesByCategory
+import entity.Genre
+
+class MovieCategoriesViewModel(
```
Same here

ZainabAlnajjar commented on feature/media_details/presentation/src/main/java/com/sanaa/presentation/screen/movie_categories/MovieCategoriesScreenInteractionListener.kt:
```diff
@@ -0,0 +1,8 @@
+package com.sanaa.presentation.screen.movie_categories
+
+interface MovieCategoriesScreenInteractionListener {
```
Same here

ZainabAlnajjar commented on feature/media_details/presentation/src/main/java/com/sanaa/presentation/screen/movie_categories/MovieCategoriesScreen.kt:
```diff
@@ -0,0 +1,133 @@
+}
+
+@Composable
+fun MovieCategoriesScreenContent(
```
Same

ZainabAlnajjar replied:
Also screen name (it give the idea you are displaying the categories of the movie, not the other way around)

ZainabAlnajjar commented on feature/media_details/presentation/src/main/java/com/sanaa/presentation/screen/actor/screen/ActorGalleryScreen.kt:
```diff
@@ -0,0 +1,123 @@
+) {
+    BackHandler(onBack = navigateBack)
+
+    val viewModel: ActorViewModel = koinViewModel { parametersOf(actorId) }
```
You can inject the viewmodel instead and pass the actor id to it directly

ZainabAlnajjar commented on feature/media_details/presentation/src/main/java/com/sanaa/presentation/screen/actor/ActorsScreenInteractionListener.kt:
```diff
@@ -0,0 +1,14 @@
+    fun onMovieClicked(id: Int)
+    fun onDismissBottomSheet()
+
+    // TODO TAKE THE ID AND SAVE MOVIE
```
I dond think this comment is needed?

ZainabAlnajjar commented on feature/media_details/presentation/src/main/java/com/sanaa/presentation/screen/actor/ActorScreenUiState.kt:
```diff
@@ -0,0 +1,18 @@
+    val topMovies: List<MovieUiModel> = emptyList(),
+    val topTvSeries: List<SeriesUiModel> = emptyList(),
+    val profileImages: List<String> = emptyList(),
+    val galleryImages: List<String> = emptyList(),
```
Add Url suffix to profile and gallery image

ZainabAlnajjar commented on domain/vod/src/main/java/entity/Genre.kt:
If you are making these changes for the presentation then this is wrong, domain shouldn't hold a value just so the ui can present it

ZainabAlnajjar commented on data/repositories/series/src/main/java/com/sanaa/series/mapper/tvSeriesMapper.kt:
```diff
@@ -44,7 +44,7 @@ fun EpisodeDto.toEntity(): Episode {
         imdbRating = voteAverage,
         durationMinutes = runtime,
-        releaseDate = LocalDate.parse(airDate),
+        releaseDate = airDate?.let { LocalDate.parse(it) } ?: LocalDate.parse("1970-01-01"),
```
If this info is nullable then you should hide it, not make fake release  date

ZainabAlnajjar commented on domain/vod/src/test/java/usecase/details/ManageActorDetailsUseCaseTest.kt:
```diff
@@ -0,0 +1,202 @@
+        val actorId = 1
+        coEvery { actorRepository.getActorDetails(actorId) } returns dummyActor
+        val result = manageActorDetailsUseCase.getActorDetails(actorId)
+        coVerify { actorRepository.getActorDetails(actorId) }
```
We don't actually need to verify that this was called, if the the resuld is equal to the expected with the same id, then that is enough of a test (and we would follow the clean code for one assert per test)

ZainabAlnajjar commented on domain/vod/src/main/java/details/usecase/tv_series/GetTvSeriesImagesUseCase.kt:
```diff
@@ -5,5 +5,9 @@ import details.repository.TvSeriesRepository
+    suspend fun execute(id: Int) = repository.getTvSeriesImages(id, COUNT)
+
+    private companion object {
+        const val COUNT = 5
```
Maximum value should be 10 as the designer specified it, same for movie images

ZainabAlnajjar commented:
Great job guys👏👏
 please consider my comments, some of them could be done in another pr if it takes long time, also you need to exclude the ui files in the kover and codecov.yml to pass the test coverage

AhmadKharfan committed:
add movie categories viewModel to Koin module

AhmadKharfan committed:
implement the movie categories screen route

AhmadKharfan committed:
provide on genre lick callback for episode details screen

AhmadKharfan committed:
change genre display name to localized string

AhmadKharfan committed:
navigate to categories screen from movie details

AhmadKharfan committed:
update movie categories screen and view model

AhmadKharfan committed:
make genres in movie details screen clickable

AhmadKharfan committed:
change type of genres in move ui module from list<string> to list<genere>

AhmadKharfan committed:
implement clickable genres in series header section

AhmadKharfan committed:
fix series details genre click navigation

AhmadKharfan committed:
enable genre navigation in series details screen

AhmadKharfan committed:
remove all unused domain layer usecases

The domain layer use cases were removed because they added unnecessary complexity and boilerplate code without providing significant value.

The use cases were simple wrappers around repository methods, leading to an extra layer of abstraction that complicated the codebase and increased development time. Removing them simplifies the architecture, reduces boilerplate, and makes the code easier to maintain and understand.

Additionally, the `Genre` enum was updated to remove the `value` property and the `fromValue` companion object function. This change simplifies the enum and aligns it with a more straightforward usage pattern, where the enum's inherent string representation is sufficient.

AhmadKharfan committed:
remove string values from genre enum

AhmadKharfan committed:
expand code coverage by adding new screen paths

AhmadKharfan committed:
enable junit platform for media details feature

AhmadKharfan committed:
adds unit tests for actor viewModel to ensure correct behavior and navigation effects

AhmadKharfan committed:
implement unit tests for episode details view model

AhmadKharfan committed:
implement unit tests for movie details view model

AhmadKharfan committed:
implement unit tests for series details view model

haideraljawasim committed:
fix a bug navagete from category screen to movie details

haideraljawasim committed:
make the images count fixed 10 for all features

haideraljawasim committed:
inject the view model in constercter for actor screens

haideraljawasim committed:
add suffix urls for actor images

hagarabobakr committed:
add test coverage for mapper

hagarabobakr committed:
update review mapper to correctly parse ID and implement tests for ReviewMapper

haideraljawasim committed:
fix faild test cases for series mange use case

hagarabobakr committed:
Add test coverage for ActorMapper

hagarabobakr committed:
add tests for mapping MoviesByCategoryDto to domain

hagarabobakr committed:
Merge remote-tracking branch 'origin/feature/navigation-details-screens' into feature/navigation-details-screens

Radwa85 committed:
add MovieCategoriesViewModel tests

haideraljawasim committed:
show only not nullable values for movie

haideraljawasim committed:
resolve conflict merging with dev

AhmadKharfan committed:
improve text expansion behavior in overview section

AhmadKharfan committed:
update review card to use expandable text component for review content

AhmadKharfan committed:
apply modifier fill maxSize to lazy column in reviews screen

AhmadKharfan committed:
Merge remote-tracking branch 'origin/feature/navigation-details-screens' into feature/navigation-details-screens

# Conflicts:
#	feature/media_details/presentation/src/main/java/com/sanaa/presentation/screen/review/ReviewsScreen.kt

ZainabAlnajjar commented on app/src/main/java/com/sanaa/novix/di/viewModelModule.kt:
```diff
@@ -0,0 +1,40 @@
+    viewModelOf(::ActorViewModel)
+    viewModelOf(::SeriesViewModel)
+    viewModelOf(::ReviewViewModel)
+    viewModelOf(::MovieCategoriesViewModel)
```
`SeriesViewModel` is duplicated

haideraljawasim committed:
fix a bug extend the text of overview

haideraljawasim committed:
delete duplacted series viewModel

AhmadKharfan committed:
handle nullable fields in movie mappers

haideraljawasim committed:
fix a bug in showing default drama on category screen

AhmadKharfan committed:
update code coverage ignore list

AhmadKharfan committed:
Merge remote-tracking branch 'origin/feature/navigation-details-screens' into feature/navigation-details-screens

hagarabobakr committed:
enhance data integrity with default values for nullable Movie DTO fields

AhmadKharfan committed:
reorder and update codecov yml exclusion list

AhmadKharfan committed:
remove unused test from series view model test

shahad-97 committed:
add review view model test

Radwa85 committed:
Merge remote-tracking branch 'origin/feature/navigation-details-screens' into feature/category-test

# Conflicts:
#	feature/media_details/presentation/src/main/java/com/sanaa/presentation/screen/movie_categories/MovieCategoriesViewModel.kt

Radwa85 committed:
fix failing MovieCategoriesViewModel test cases

AhmadKharfan committed:
Merge pull request #189 from SanaaTeam/feature/review-test

Feature/review test

AhmadKharfan committed:
remove junit 4

haideraljawasim committed:
resolve conflict merging with dev

haideraljawasim committed:
fix imports for remote data source test

haideraljawasim committed:
fix unit test for view model

haideraljawasim committed:
resolve conflict merging with development

haideraljawasim committed:
delete unwanted tests for details

haideraljawasim committed:
fix tests related to deatails data source

mostafa-n3ma committed:
Merge remote-tracking branch 'origin/development' into feature/navigation-details-screens

mostafa-n3ma committed:
fix : test debug bug

mostafa-n3ma committed:
Merge branch 'development' into feature/navigation-details-screens

# Conflicts:
#	codecov.yml
#	domain/vod/src/test/java/usecases/search/SearchMoviesUseCaseTest.kt
#	domain/vod/src/test/java/usecases/search/SearchTvSeriesUseCaseTest.kt

mostafa-n3ma committed:
fix: ReviewViewModelTest.kt

mostafa-n3ma committed:
update lint settings

mostafa-n3ma committed:
disabling lint operations
